### PR TITLE
977 redux

### PR DESCRIPTION
**Pull request by JMcGavigan32**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md`
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue so that it closes after successful merging. 
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done!  Much love to your accomplishment!

### CODE REVIEW

**Comment by JMcGavigan32** [comment]
Hello @bms63 @bundfussr. There are still some things that need to be updated  per your suggestions, but I thought I would create the pull request so you could see progress. I know we are getting `#> Error in assert_vars(required_vars, optional = TRUE): `required_vars` must be a list of unquoted variable names, e.g. `vars(USUBJID, VISIT)` which needs to be rectified.
Will get this finished tomorrow.

**Comment by bms63** [comment]
Okay I am going to close Pull Request #1012.   Just to confirm: The issue was that the original author created their branch from `main` and not from `devel`?

**Comment by bms63** [comment]
Hey @JMcGavigan32  make sure to do `devtools::test()`.  I found something in your code which I think is a mistake - maybe from styler.

Use the Find in Files thing I showed you to find examples of how other folks dealt with `assert_data_frame()` with how it is set up for this function.  For `devive_derive_param()`  looks promising.

Also, unsure about this direct call of `assert_symbol(enquo(XYZ))` directly in the function call.

**Comment by JMcGavigan32** [comment]
Hi @bms63. For sure will look at these tomorrow, just wanted to get the pull request in there before the end of the day

**Comment by JMcGavigan32** [comment]
> Hey @JMcGavigan32 make sure to do `devtools::test()`. I found something in your code which I think is a mistake - maybe from styler.
> 
> Use the Find in Files thing I showed you to find examples of how other folks dealt with `assert_data_frame()` with how it is set up for this function. For `devive_derive_param()` looks promising.
> 
> Also, unsure about this direct call of `assert_symbol(enquo(XYZ))` directly in the function call.

Hi @bms63 I think I have sorted out the `assert_data_frame()` error by calling `assert_vars(by_vars)` before it. 

I have also removed `style_text()` which was causing some issues too. I believe it was an incorrect implementation of this: 

image


Regarding `assert_symbol(enquo(XYZ))`, this was my interpretation of a comment made by Stefan: 

image

**Comment by bundfussr** [comment]
@JMcGavigan32 , sorry, I did not mean to add it as default value. This will not work because then only the default value is quoted and checked but not non-default values provided by the users.

You should set `age = AGE` for the default value and then in the function body add
```
age <- assert_symbol(enquo(age))
```

**Comment by JMcGavigan32** [comment]
> @JMcGavigan32 , sorry, I did not mean to add it as default value. This will not work because then only the default value is quoted and checked but not non-default values provided by the users.
> 
> You should set `age = AGE` for the default value and then in the function body add
> 
> ```
> age <- assert_symbol(enquo(age))
> ```

No apologies needed @bundfussr! I am such a novice. Thank you so much for taking the time to clarify that. I will implement that now.

**Comment by bms63** [comment]
@JMcGavigan32 Hey Jack.  I think we will need to get this into the next release.  We would like to freeze the devel branch to to do PR into main.  Unless you are really close to finishing

**Comment by JMcGavigan32** [comment]
That's alright with me! @bms63

**Comment by bms63** [comment]
@thomas-neitmann @millerg23  do either of you have time to look over?  Alice created this and Jack took over, but wasn't able to finish .  ::((Hoping to get this one in.  I think I fixed all the issues.

File: R/compute_framingham.R [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    sex == "M" ~
+    1 - (0.88936^exp((3.06117 * log(age))
+    + (1.12370 * log(chol))
+      - (0.93263 * log(cholhdl))
+      + (1.93303 * log(if_else(trthypfl == "N", sysbp, 1)))
+      + (1.99881 * log(if_else(trthypfl == "Y", sysbp, 1)))
+      + (0.65451 * (if_else(smokefl == "Y", 1, 0)))
+      + (0.57367 * (if_else(diabetfl == "Y", 1, 0)))
+      - 23.9802))
+  )
```
Comment (thomas-neitmann) [inline R/compute_framingham.R:144]: Please use `if_else()` instead of `case_when()`.

File: R/derive_param_framingham.R [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    required_vars =
+      quo_c(
+        vars(!!!by_vars, PARAMCD, AVAL),
+        enquo(age),
+        enquo(sex),
+        enquo(smokefl),
+        enquo(diabetfl),
+        enquo(trthypfl)
+      )
+  )
```
Comment (thomas-neitmann) [inline R/derive_param_framingham.R:220]:
```suggestion
  assert_data_frame(
    dataset,
    required_vars = quo_c(
      vars(!!!by_vars, PARAMCD, AVAL),
      enquo(age),
      enquo(sex),
      enquo(smokefl),
      enquo(diabetfl),
      enquo(trthypfl)
    )
  )
```

File: R/derive_param_framingham.R [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+#' + (log(TotalChol) * TotalCholFactor)
+#' + (log(CholHDL) * CholHDLFactor) \\
+#' + (log(SysBP) * SysBPFactor) + Smoker
+#' + Diabetes Present - AvgRisk}
+#'
+#' \deqn{Risk = 100 * (1 - RiskPeriodFactor^exp(RiskFactors))}
+#'
+#' @author
+#' Alice Ehmann
+#' Jack McGavigan
```
Comment (thomas-neitmann) [inline R/derive_param_framingham.R:132]: Please add yourself as author. Also Jack should be listed in DESCRIPTION.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]
Looks good to me! Just some minor things.

**Comment by millerg23** [comment]
@bms63 only concern I have is, the lab units rerquested are `mg/dL` which is the Conventional Unit for Cholesterol, however we normally use SI units, which is `mmol/L`. For the absolute conversions, and for grading we porvide only SI units for acceptable units. Maybe worth a discussion to see if we should provide both CV and SI units, or SI units only.

**Comment by bms63** [comment]
Hey @millerg23 thanks for taking a look.  The paper that is used to build this function uses the `mg/dl` units.  So...a little uncertain, but If someone want to use this function would they not first convert their SI data to the conventional unit to make use of this function?  I feel like this function is a bit niche.  

https://www.ahajournals.org/doi/pdf/10.1161/CIRCULATIONAHA.107.699579

**Comment by millerg23** [comment]
They might not even have to do that, usually we have 2 sets of lab results, LBSTRESN for SI values, and LBCVRESN the holds the conventional unit. So they could just set `AVAL = LBCVRESN` and function would work. Its probably OK then if its quite niche, but wouldn't add it into the ADLB template, as it could get confusing.
I will approve, but maybe worth discussing in our lab team, if we should provide grading for both CV and SI units?

**Review by millerg23 — APPROVED** [review]

File: R/compute_framingham.R (hunk 1 above)
Comment (thomas-neitmann) [reply R/compute_framingham.R:144]: How so? It's one of two condition, right? Either you're male or female, so `if_else()` seems like the way to go. I'm pretty sure we have that in our programming strategy to use `if_else()` over `case_when()` when there are only two possibilities.

Comment (bms63) [reply R/compute_framingham.R]: You are totally right!!  Updated.

**Review by thomas-neitmann — APPROVED** [review]

**Pull request merged**